### ISSUE TITLE
Proposal for Privacy-Enhanced Display of NIP-05 Verification Status

### ISSUE THREAD

**Issue by Arvin21M** [nostr-protocol/nips#843]
:zap:

I am a dedicated and experienced user of Nostr, having explored and interacted with various Nostr clients both on mobile and web platforms extensively from a UX/UI perspective. While I lack the coding expertise to implement protocol changes, I have come across a potential area of enhancement while attempting to obtain my NIP-05 verification, which I believe could significantly improve user privacy without compromising the core objectives of verification, if it is possible.

**Issue:**
The current NIP-05 verification process displays the domain URL on a user's profile upon verification. This visibility, while transparent, could pose privacy concerns for users who wish to authenticate their profiles without disclosing their domain URLs, but who also do not want to use "off-the-shelf" verification providers.

**Proposed Enhancement:**
Modifications to the NIP-05 protocol to enable a privacy-enhanced verification display. Ideally, the "system" would perform the public key to DNS mapping in the background, issue a verification status upon satisfying the verification parameters, and display the verification status on the user's profile without revealing the domain URL.

**Benefits:**

1. Catering to a broader range of privacy preferences among Nostr users.
2. Enhancing user comfort and trust in the verification process.
3. Potentially attracting more users to the platform who value more granular privacy like this.

I am eager to engage with the community and technical experts on this proposal. Your feedback, insights, and suggestions would be highly invaluable in refining this idea and exploring its feasibility.

Moreover, I am open to collaborating with technical members of the community, in any way that I am able, who might be interested in taking this proposal forward.

Thank you.

**Comment by erskingardner** [comment]
I don't think that this makes sense... 

1. To start with, NIP-05 isn't really a strong "verification". It's just a way to prove that you have the ability to change something on a publicly available webserver. 
2. The whole point of NIP-05 is that it has to be public so that clients/users/etc. can use an out-of-nostr system to build more confidence that you are who you say you are. Again, it is, by no means, a strong verification that you're definitely the right person. It's just a hint that you might be.

**Comment by Arvin21M** [comment]
Thanks JeffG (@erskingardner)!

I understand... My suggestion seems to step back/down from NIP-05. If a client verifies a user via NIP-05 but only shows a "verified" badge, other users can't independently verify. 

This proposal does not enhance NIP-05, but could it inspire a new NIP for a different user authentication method? 

Displaying a verified badge without the domain name could be a middle ground for users preferring some verification without disclosing their domain or using external verification services. Certainly no urgent matter here, but seems nice to have.

**Comment by BrutusBondBTC** [comment]
Nip 05 can be fairly private if you host you use github or some other external web hosting service that does not require KYC info.

 Just use a burner email.

It would be nice to have another option, however, like a PGP fingerprint. I understand PGP is not the best option from a UX perspective, but it's cryptographically solid and could prove who the real Adam Back is with a high degree of certainty.

I'm thinking about a signed PGP message + a simple passphrase 

-----BEGIN PGP MESSAGE-----

jA0ECQMCjjQvTlBKaVpg0pUBi6uGf/sAQbmD4lLqSWNWk7xOf42UoSysap3T4qU6
h3Ls4l20zL/yAANfsGcBWzQG4LpMh/oXabLY+mlXTt/aWb3VnPvA4uujJTtDAdiH
aCMBvQ1+239K8F/mnGzBflXz0zywyyp03wSwKdo4Z0R6lPXIKiMHOyYaKUiA3Mna
1DDvtJ9Yc6PhPvGWe8wfRrqdCnpHrg==
=GJm7
-----END PGP MESSAGE-----


In this example, my passphrase is:
nostr

But the message is signed with my PGP key.

I'm not a real cryptographer, but could something like this work?

**Comment by erskingardner** [comment]
> I understand... My suggestion seems to step back/down from NIP-05. If a client verifies a user via NIP-05 but only shows a "verified" badge, other users can't independently verify.

The domain would have to show up somewhere public, like your `kind:0` metadata so I don't think that you could keep it "secret". Clients could decide to show the domain or not in their own UIs, but again, it's not secret.

> I'm thinking about a signed PGP message + a simple passphrase

Yeah – I guess something like this could be added. The idea of using multiple layers of cryptographic signatures would give other users more confidence that they are talking to the real version of you. That said, I'm not familiar enough with the pros/cons of PGP to know if that's a good solution or not.

@staab @pablof7z @fiatjaf thoughts?

**Comment by pablof7z** [comment]
This makes no sense to me. You use a NIP-05 explicitly to be found under a more human friendly identifier. It being public is the goal.

Wrt tying to a PGP pubkey, PGP has even less network effects than nostr and is equally weird-looking which is in part the point of keybase.

The issue this refers to stems from a confusion clients introduce by using the badge which looks to similar to twitter's but it means something different

TLDR; concept NACK 😅

**Comment by Arvin21M** [comment]
@pablof7z 

> This proposal does not enhance NIP-05, but could it inspire a new NIP for a different user authentication method?

An alternative method where users don't publicly name assets they own/control OR don't use/name 3rd-party verification services in order to "prove" that it is more likely than not they are who they say they are.

**Comment by Arvin21M** [comment]
All right, in light of two responses starting off with "this doesn't make sense," I am going to close this issue with this comment. We still need simpler ways to authenticate without revealing ownership of assets available/visible to the general public.

![image](https://github.com/nostr-protocol/nips/assets/141419722/9948415d-6eb5-47d5-b99e-a380c24e3a3c)
